src/AliasRecord.php

```diff
@@ -71,7 +71,7 @@ public function hasRoot()
     public function root()
     {
         $root = $this->get('root');
-        if ($this->isLocal()) {
+        if (!is_null($root) && $this->isLocal()) {
```

Comment:
It would be better to move this null check into `FsUtils::realpath()` rather than make every client do the same check.

Reply:
Is this what you're thinking?

```diff
index b5741f2..f5e0b68 100644
--- a/src/Util/FsUtils.php
+++ b/src/Util/FsUtils.php
@@ -10,16 +10,20 @@ class FsUtils
      * Returns canonicalized absolute pathname.
      *
      * The difference between this and PHP's realpath() is that this will
-     * return the original path even if it doesn't exist.
+     * return the original path even if it doesn't exist, and also accounts for
+     * a situation where $path may be null.
      *
-     * @param string $path
+     * @param string|null $path
      *   The path being checked.
      *
-     * @return string
-     *   The canonicalized absolute pathname.
+     * @return string|null
+     *   The canonicalized absolute pathname, or null if $path was null.
      */
     public static function realpath($path)
     {
+        if (is_null($path)) {
+            return null;
+        }
         $realpath = realpath($path);
         return $realpath ?: $path;
     }
```

It seems a bit strange to me to accept a null there and then return a null, but I'm happy to make the change if you'd prefer it that way.